modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/CompositeArtifactExtensionManager.java

```diff
@@ -132,12 +132,10 @@ public ConfigurationInstance getConfiguration(String configurationProviderName,
   public Optional<ConfigurationProvider> getConfigurationProvider(ExtensionModel extensionModel, ComponentModel componentModel) {
     Optional<ConfigurationProvider> configurationModel =
         childExtensionManager.getConfigurationProvider(extensionModel, componentModel);
-
-    if (!configurationModel.isPresent()) {
+    if (!configurationModel.isPresent() && !childExtensionManager.getExtension(extensionModel.getName()).isPresent()) {
```

Comment:
pretty sure this breaks backwards compatibility when a config is defined at the domain level

Reply:
You can have the case in which both the child and parent both define an extension, but the child uses it implicitly while the parent uses it explicitly. In that case, one of two things can happen:

1) This change is superflous as it ends up looking into the parent anyway
2) it returns an empty Optional when before it would had returned the parents provider, therefore breaking backwards compatibility

Reply:
The code has been refactored and the CompositeArtifactExtensionManager is no longer modified.